src/jit/lsrabuild.cpp

```diff
@@ -2560,13 +2563,16 @@ void LinearScan::BuildDefsWithKills(GenTree* tree, int dstCount, regMaskTP dstCa
     {
         buildKillPositionsForNode(tree, currentLoc + 1, killMask);
 #if FEATURE_PARTIAL_SIMD_CALLEE_SAVE
-        if (enregisterLocalVars && ((killMask & RBM_FLT_CALLEE_TRASH) != RBM_NONE))
+        if ((killMask & RBM_FLT_CALLEE_TRASH) != RBM_NONE)
         {
             // Build RefPositions for saving any live large vectors.
             // This must be done after the kills, so that we know which large vectors are still live.
             VarSetOps::AssignNoCopy(compiler, liveLargeVectors,
                                     buildUpperVectorSaveRefPositions(tree, currentLoc + 1, killMask));
-            doLargeVectorRestore = true;
+            if (enregisterLocalVars && !VarSetOps::IsEmpty(compiler, liveLargeVectors))
```

Comment:
The same comment about adding  `assert(enregisterLocalVars)` under `if`.

Reply:
Currently the lclVar sets are not initialized if `enregisterLocalVars` is false, so we can't move the condition to an assert (i.e. we need to check `enregisterLocalVars` before we test whether the set is empty).